zarp/run/snakemake.py

```diff
@@ -0,0 +1,49 @@
+"""Module for running snakemake."""
+
+import subprocess
+import os
+
+from zarp.config.models import Run
+
+
+class SnakemakeExecutor:
+    """Run snakemake with system calls."""
+
+    def __init__(self, run: Run) -> None:
+        self.run_dict = run.dict()
+        self.success = None
+
+    def prepare_run(self, snkfile: str, workdir: str) -> None:
+        """Configure list of strings for execution.
+        """
+        run_list = ["snakemake"]
+        # Snakemake config
+        run_list.extend(["--snakefile", snkfile])
+        run_list.extend(["--cores", str(self.run_dict["cores"])])
+        # run_list.extend(["--config", f"workdir={workdir}"])
+        # execution profile
+        if self.run_dict['execution_profile'] == "local-conda":
+            prof = ["--profile", os.path.join("submodules", "zarp", "profiles", "local-conda")]
+            run_list.extend(prof)
+        self.run_list = run_list
```

Comment:
This attribute is not defined in the constructor. Also, it's not good practice to modify class attributes silently within methods - it's the equivalent of modifying global variables in functions ("side effects"). Best is to define an empty run list in the constructor, and have this method return `run_list`, then assign it to `self.run_list` in the calling context.

Remember: be careful when assigning to `self.some_attribute` in a method - or your method will have side effects! I only really became conscious of that recently myself during a hackathon, and HTSinfer, for example, is full of this stuff - but it's already biting my back because it makes the code quite fragile. The exception to this anti-pattern are setter methods by the way - but they are deliberately short and explicit, so that it reduces the risks of inadvertently suffering from side effects.